### ISSUE TITLE
New rule: Assignment replaceable with operator assignment

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#1420]
This is the first rule from WebStorm Inspectors (rollup) #667.

> This inspection reports instances of assignment operations in JavaScript content which can be replaced by operator-assignment. Code using operator assignment may be clearer, and theoretically more performant.

Valid patterns:

``` js
x += y;
x = y - x;
x.y = x['y'] + z; // dot-notation error, but not this
x = (x * y) * z;
```

Invalid patterns:

``` js
x = y + x;
x = x / y;
x.y = x.y & z;
x[0] = y * x[0];
```

Please correct these guesses:
- Name `operator-spacing`
- On by default
